tests/D2L.CodeStyle.Analysis.UnitTests/UtilsTests.cs

```diff
@@ -0,0 +1,41 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Text;
+using NUnit.Framework;
+
+namespace D2L.CodeStyle.Analysis {
+
+    internal sealed class UtilsTests {
+
+        private readonly Utils m_utils = new Utils();
+
+        [Test]
+        public void IsGeneratedCodefile_NotCSharpFile_False() {
+            const string file = @"proj\random.js";

```

Comment:
Agreed, but having the '.generated' in the path means that the order of the check in code doesn't matter if the code changes.